test/test_relationship.rb

```diff
@@ -0,0 +1,17 @@
+require_relative "test_helper"
```

Comment:
I would have expected to auto-assignment tests. Tests where you provide the objects and the class sets the type itself. Is there a reason why you didn't make such tests?